e2e/tests/visual-a11y/styling.visual.spec.js

```diff
@@ -0,0 +1,301 @@
+
+/**
+ * This test is dedicated to test styling of various plugins
+ */
+
+const { test, expect } = require('../../pluginFixtures');
+const { createDomainObjectWithDefaults } = require('../../appActions');
+
+test.describe('Flexible Layout styling', () => {
+  let sineWaveObject;
+  let sineWaveObject2;
+  let treePane;
+  let sineWaveGeneratorTreeItem;
+  let sineWaveGeneratorTreeItem2;
+  let stackedPlotTreeItem;
+  let stackedPlot;
+  let flexibleLayout;
+  test.beforeEach(async ({ page }) => {
+    await page.goto('./', { waitUntil: 'domcontentloaded' });
+
+    // Expand the 'My Items' folder in the left tree
+    await page.locator('.c-tree__item__view-control.c-disclosure-triangle').click();
+
+    // Create a few Sine Wave Generators
+    sineWaveObject = await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator'
+    });
+
+    sineWaveObject2 = await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator'
+    });
+
+    // Define the Sine Wave Generator items
+    treePane = page.getByRole('tree', {
+      name: 'Main Tree'
+    });
+    sineWaveGeneratorTreeItem = treePane.getByRole('treeitem', {
+      name: new RegExp(sineWaveObject.name)
+    });
+    sineWaveGeneratorTreeItem2 = treePane.getByRole('treeitem', {
+      name: new RegExp(sineWaveObject2.name)
+    });
+
+    // Create a Stacked Plot
+    stackedPlot = await createDomainObjectWithDefaults(page, {
+      type: 'Stacked Plot'
+    });
+
+    // Add the Sine Wave Generators to the Stacked Plot
+    await sineWaveGeneratorTreeItem.dragTo(page.locator('.c-plot--stacked.holder'));
+    await sineWaveGeneratorTreeItem2.dragTo(page.locator('.c-plot--stacked.holder'));
+    await page.locator('button[title="Save"]').click();
+    await page.locator('text=Save and Finish Editing').click();
+
+    stackedPlotTreeItem = treePane.getByRole('treeitem', {
+      name: new RegExp(stackedPlot.name)
+    });
+
+    // Create a Flexible Layout
+    flexibleLayout = await createDomainObjectWithDefaults(page, {
+      type: 'Flexible Layout'
+    });
+
+    // Add the Sine Wave Generators and the Stacked Plot to the Flexible Layout
+    await sineWaveGeneratorTreeItem.dragTo(page.locator('.c-fl__container').first());
+    await sineWaveGeneratorTreeItem2.dragTo(page.locator('.c-fl__container').first());
+    await stackedPlotTreeItem.dragTo(page.locator('.c-fl__container').nth(1));
+
+    await page.locator('button[title="Save"]').click();
+    await page.locator('text=Save and Finish Editing').click();
+  });
+
+  test('selecting a flexible layout column hides the styles tab', async ({ page }) => {
+    // Edit Flexible Layout
+    await page.locator('[title="Edit"]').click();
+    // Expect to find a styles tab
+    let stylesTab = await page.locator('.c-inspector__tab[title="Styles"]').count();
+    expect(stylesTab).toBe(1);
+
+    // Select flexible layout column
+    await page.locator('.c-fl-container__header').first().click();
```

Comment:
these are great tests. I need to update the locator strategy and it may involve some changes to the flexibleLayout component props